main.go

```diff
@@ -559,18 +562,30 @@ func DeleteS3Backups(backupTime time.Time, retentionPeriod time.Duration, svc *m
 		}
 		// only parse backup file names that matches *_etcd format
 		if re.MatchString(object.Key) {
-			backupTime, err := time.Parse(time.RFC3339, strings.Split(object.Key, "_")[0])
+			filename := object.Key
+
+			if len(bc.Folder) != 0 {
+				// example object.Key with folder: folder/timestamp_etcd.zip
+				// folder and separator needs to be stripped so time can be parsed below
+				log.Debugf("Stripping [%s] from [%s]", fmt.Sprintf("%s/", prefix), object.Key)
+				filename = strings.TrimPrefix(object.Key, fmt.Sprintf("%s/", prefix))
+			}
+			log.Debugf("Filename: [%s]", filename)
```

Comment:
prolly nit: since we're logging filename now, can we use filename consistently everywhere below in place of `object.Key`?

Reply:
`object.Key` contains the full path (including folder) which is needed for the delete, I added a comment to clarify that